framework/caching/FileCache.php

```diff
@@ -141,6 +141,12 @@ protected function setValue($key, $value, $duration)
         if ($this->directoryLevel > 0) {
             @FileHelper::createDirectory(dirname($cacheFile), $this->dirMode, true);
         }
+        // If ownership differs the touch call will fail, so we try to
+        // rebuild the file from scratch by deleting it first
+        // https://github.com/yiisoft/yii2/pull/16120
+        if (is_file($cacheFile) && fileowner($cacheFile) !== posix_geteuid()) {
```

Comment:
`posix_geteuid()` may not be available. For example, it's not available under Windows.

Reply:
If so this library needs an [AppVeyor](https://www.appveyor.com/) build.

It would be easy to add a `function_exists('posix_geteuid')` here, but I don't like making changes that aren't tested.

Reply:
Merged AppVeyor. It fails in some tests: https://ci.appveyor.com/project/samdark/yii2

Reply:
I think you need to merge with master in order for build to be triggered...

Reply:
https://ci.appveyor.com/project/samdark/yii2/build/dev-6#L274

Now that I have a failing test, I'll fix the code